website/static/js/comment.js

```diff
@@ -62,6 +115,54 @@ var exclusifyGroup = function() {
     }
 };
 
+ko.subscribable.fn.convertHtmlToMarkdown = function(commentContent) {
+    return ko.computed(function() {
+        var content = commentContent() || '';
+        var regex = /<span[^>]*?data-atwho-guid="([a-z\d]{5})"[^>]*?>((@|\+)[\w\s]+)<\/span>/;
+        var matches = content.match(/<span[^>]*?data-atwho-guid="([a-z\d]{5})"[^>]*?>((@|\+)[\w\s]+)<\/span>/g);
+        if (matches) {
+            for (var i = 0; i < matches.length; i++) {
+                var match = regex.exec(matches[i]);
+                var guid = match[1];
+                var mention = match[2];
+                var url = '/' + guid + '/';
+                content = content.replace(match[0], '['+ mention + '](' + url + ')');
+            }
+        }
+        // '&#13;&#10;' is the character entity reference for '\r\n'
+        // '\r\n' is treated differently and breaks conversion from markdown to html
+        content = content.replace(/<span[^>]*?>/g, '')
+            .replace(/<\/span>/g, '')
+            .replace(/&nbsp;/g, ' ')
+            .replace(/<br>/g, '&#13;&#10;');
+        return content;
+    }, this);
+};
+
+ko.subscribable.fn.convertMarkdownToHtml = function(commentContent) {
+    return ko.computed(function() {
+        var content = commentContent();
+        var regex = /\[(@|\+)(.*?)\]\(\/([a-z\d]{5})\/\)/;
+        var matches = content.match(/\[(@|\+)(.*?)\]\(\/([a-z\d]{5})\/\)/g);
+        if (matches) {
+            for (var i = 0; i < matches.length; i++) {
+                var match = regex.exec(matches[i]);
+                var atwho = match[1];
+                var guid = match[3];
+                var mention = match[2];
+
+                content = content.replace(
+                    match[0],
+                    '<span class="atwho-inserted" contenteditable="false" data-atwho-guid="' +
+                        guid + '" data-atwho-at-query="' + atwho + '">' +
+                        atwho + mention + '</span>'
+                );
+            }
+        }
+        return content.replace(/\r\n/g, '<br>');
+    }, this);
+};
+
```

Comment:
We should have some tests for these two functions, because regular expressions are complex and error prone.